### ISSUE TITLE
Register 0.2

### ISSUE THREAD

**Issue by mcabbott** [FluxML/Optimisers.jl#52]
What needs to happen before we tag this, and depend on it alla: https://github.com/FluxML/Flux.jl/pull/1864?

**Comment by cossio** [comment]
Just do it? There are only two packages which depend on Optimisers and they have strict Compat so they won't be affected. So it would be nice just to see if tests at https://github.com/FluxML/Flux.jl/pull/1864 pass.

**Comment by darsnack** [comment]
I took a pass over the current codebase. It appears to be at a good stage to release and start integrating in Flux.

**Comment by ToucheSir** [comment]
> Just do it? There are only two packages which depend on Optimisers and they have strict Compat so they won't be affected. So it would be nice just to see if tests at [FluxML/Flux.jl#1864](https://github.com/FluxML/Flux.jl/pull/1864) pass.

Just a note that there are at least a couple unregistered packages which depend on Optimisers. The total number is still small, of course.

The bigger question to answer before cutting a 0.2 is whether we have any imminent breaking changes coming down the pipe. Cutting a 0.3 shortly after 0.2 wouldn't be ideal. To my knowledge https://github.com/FluxML/Optimisers.jl/pull/42 is the only potentially breaking in-flight PR.

**Comment by mcabbott** [comment]
Yes to not wanting too many breaking releases. I am not sure #42 is breaking, since the cases it changes are now an error. But I think #33 counts.

**Comment by ToucheSir** [comment]
I didn't include #33 because it didn't seem like we were anywhere close to a consensus opinion on that. #42 is trickier because AIUI it was possible to use array wrappers untied before, but with the PR they'd lead to a different (non-duck typed) return value.

**Comment by mcabbott** [comment]
> was possible to use array wrappers untied before

I think you mean that Transpose would prevent it from noticing shared weights, before. And thus you won't get an error, but `update!` still write into both. This is the same as Flux's behaviour right now. I'm not sure whether that can be called a bug & fixed, or must be called a change. 

For Functors alone, changing not to regard Transpose as a leaf node seems more obviously a change. There may be other uses besides such optimisation. Likewise changing not to cache `isbits` things.

**Comment by ToucheSir** [comment]
> I think you mean that Transpose would prevent it from noticing shared weights, before. And thus you won't get an error, but `update!` still write into both. This is the same as Flux's behaviour right now. I'm not sure whether that can be called a bug & fixed, or must be called a change.

Yes, I don't know where to place it since this is what PyTorch does too (and AFAICT people are using it blissfully unaware of the consequences in the wild).

**Comment by mcabbott** [comment]
Dropout, BatchNorm and friends should already have restricted `Flux.trainable`. And Conv should only store integers. But making Functors not tie numbers doesn't look close. 

So maybe we should tag Optimisers 0.2 now, which Flux 0.13 can depend on. 

Then Functors 0.3 + Optimisers 0.3 will have #42 , which needs https://github.com/FluxML/Functors.jl/pull/33 which is breaking. And (I hope) a more clever cache allowing scalars, once we find the brain cycles to get that right.

**Comment by ToucheSir** [comment]
Makes sense to me, would you like to do the honours?

**Comment by mcabbott** [comment]
@JuliaRegistrator register
